Azure/sonic-buildimage
# [multi-asic][vs]: Add new multi-asic vs hwsku with four asics

**- Why I did it**
- Current mutli-asic vs hwsku consists of 6 asics with each asic having 32 interfaces. When bringing this up, below issue was seen:
- When all 32 interfaces(sonic interfaces and linux interface) are set to 9100 mtu, DMA error is seen "DMA: Out of SW-IOMMU space for 4096 bytes at device 0000:06:03.0" which can be fixed by updating swiotlb=65536 in /host/grub/grub.cfg .

In order to keep multi-asic VS lighter and easier to bring up and test, new hwsku 'msft_four_asic_vs' is added to represent 4-asic hwsku with 2 frontend asics and 2 backend asics and each asic having 8 interfaces interconnected by port-channels.

**- How I did it**
- Add msft_four_asic_hwsku directory to have the right number of directories (4) and update port_config.ini and lanemap.ini files to include 8 ports information.
- Add topology.sh script to create the internal asic-asic connectivity.

**- How to verify it**
- Update asic.conf with the 4 asic information as below and build sonic-vs.img:
NUM_ASIC=4
DEV_ID_ASIC_0=0
DEV_ID_ASIC_1=1
DEV_ID_ASIC_2=2
DEV_ID_ASIC_3=3
- Modify sonic_multiasic.xml to have 8 front panel interfaces.
- create virtual switch using "sudo virsh sonic_mutliasic.xml" command.
- Start topology service and Load config_db files for switch and each asic.
- Ensure that that all internal interfaces and port_channels are coming up.
multi-asic vs testbed:
- Bring up mutli-asic VS testbed with a multi-asic image(asic.conf updated to 4 asics) and using t1-lag topology.
 ./testbed-cli.sh -t vtestbed.csv -m veos_vtb -k ceos add-topo vms-kvm-four-asic-t1-lag password.txt
 - Load minigraph/config_dbs.
 - Ensure all internal and external interfaces come up.
- No change on single asic vs.

**- Which release branch to backport (provide reason below if selected)**




**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**
